fe/src/components/ProductCard/ProductCard.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Flex,
+  Box,
+  Image,
+  useColorModeValue,
+  Icon,
+  chakra,
+  Tooltip,
+} from '@chakra-ui/react';
+import {
+  AddIcon,
+  CheckIcon,
+  CloseIcon,
+  HamburgerIcon,
+} from '@chakra-ui/icons';
+
+// @todo - Remove this interface after we have a type for Product as "single source of truth".
+interface Product {
+  id: number,
+  name: string,
+  description: string,
+  isAvailable: boolean,
+  imageUrl: string,
+}
+
+function ProductCard({ product } : { product: Product }) : JSX.Element {
```

Comment:
I think all components that have props should clearly state that they are React components. If you don't do so it will cause super confusing type errors later on.

```jsx
import * as React from 'react'; // <- React should always be imported like this

interface Props {
  product: Product
}

const ProductCard: React.FC<Props> = ({ product }) => { ... }
```